### PR TITLE
Use getLegendsData options to change the others label

### PR DESCRIPTION
**Pull request by elenatorro**
# Context

We should make use of the 'othersLabel' parameter option we have in the `getLegendsData` method

* [x] Change examples
* [x] Update guide cc @makella

### CODE REVIEW

**Comment by makella** [comment]
Examples to update:
- [ ] https://carto.com/developers/carto-vl/examples/#example-point-legend
- [ ] https://carto.com/developers/carto-vl/examples/#example-line-legend
- [ ] https://carto.com/developers/carto-vl/examples/#example-image-legend

Guide maps to update:
- [ ] https://carto.com/developers/carto-vl/examples/maps/guides/add-legends/step-4.html
- [ ] https://carto.com/developers/carto-vl/examples/maps/guides/add-legends/step-5.html
- [ ] https://carto.com/developers/carto-vl/examples/maps/guides/add-legends/step-8.html

- [ ] Update guide text
- [ ] Last map in the guide is pointing to wrong step so I'll update that too

**Comment by makella** [comment]
Ah! thanks @elenatorro for updating the data driven viz examples. So adding those:

- [x] examples/guides/data-driven-viz-1/step-8.html
- [x] examples/guides/data-driven-viz-1/step-9.html
- [x] examples/guides/data-driven-viz-2/step-2.html

**Comment by makella** [comment]
@elenatorro do you mind checking

* examples/maps/guides/add-legends/step-8.html

The map isn't loading properly on this branch... not exactly sure what the issue is! 

thanks

**Comment by makella** [comment]
@elenatorro sorry I am getting a little confused here!

Ok, so this step still has CARTO_VL_OTHERS
http://127.0.0.1:8080/examples/guides/add-legends/step-5.html

That is the step where you can overwrite the `CARTO_VL_OTHERS` to something else, in the guide it is `Other weather` is that still possible?

In http://127.0.0.1:8080/examples/guides/add-legends/step-4.html, the `CARTO_VL_OTHERS` should show since it is the default.

Same case here:
http://127.0.0.1:8080/examples/guides/data-driven-viz-1/step-8.html

And here:
http://127.0.0.1:8080/examples/guides/data-driven-viz-2/step-2.html

Basically, I think I am really confused and not sure how to overwrite that CARTO_VL_OTHERS. Is that still possible? (hope so!)

**Comment by makella** [comment]
ok, I see the syntax here:
https://github.com/CartoDB/carto-vl/pull/1282/commits/30dbd5cbc8d31000daf707038c6e2fef4bfaa42b#diff-ee6881df729955a85735efc6193ea025

let me try

**Comment by makella** [comment]
weird! I see that you changed step-5, but I don't see the changes on the map. Not sure what's up, but maybe easiest to talk this through so I stop confusing myself and adding so much noise to the PR :)

**Comment by elenatorro** [comment]
Good catches, @makella 👌 

We've fixed the CARTO_VL_OTHERS default label replacement in this PR https://github.com/CartoDB/carto-vl/pull/1280 because it was related with how we build the legends data for the histograms. In the step-5 you mention, it's not working cause it is a bug, but we've already fixed that here:
https://github.com/CartoDB/carto-vl/pull/1280/files#diff-49708e6aada6e09801368e7411650a0b

Let's wait until we merge it to check everything works fine 👍

**Comment by makella** [comment]
@elenatorro took a detailed look through and everything LGTM!

Since we modified the guide in this branch, I also made minor legend guide fix. We had some of the example maps at the end linking to the wrong maps so I went ahead and updated those too.

**Comment by elenatorro** [comment]
Awesome, nice work! LGTM

File: examples/legends/category-line-legend.html [hunk 1]
```diff
@@ -64,16 +64,14 @@ <h1>San Francisco Streets</h1>
...
       });
 
       let colorLegendList = '';
       colorLegend.data.forEach((legend) => {
         const colorHex = rgbToHex(legend.value);
-        const type = legend.key;
-        const entry = type !== 'CARTO_VL_OTHERS' ? type : 'Others';
         colorLegendList +=
-          `<li><span class="${viz.geometryType}-mark" style="border-color:${colorHex};"></span> <span>${entry}</span></li>\n`;
+          `<li><span class="${viz.geometryType}-mark" style="border-color:${colorHex};"></span> <span>${legend.key}</span></li>\n`;
```
Comment (VictorVelarde) [inline examples/legends/category-line-legend.html]: Nice refactor
